tests/nnstreamer_plugins/unittest_plugins.cpp

```diff
@@ -2992,6 +2993,352 @@ TEST (test_tensor_filter, reopen_tflite_02_p)
+
+  /* get output buffer */
+  out_buf = gst_harness_pull (h);
+  EXPECT_EQ (gst_buffer_n_memory (out_buf), 1U);
+  EXPECT_EQ (gst_buffer_get_size (out_buf), out_size);
+  gst_buffer_unref (out_buf);
+
+  /* set second model file */
+  gst_harness_set (h, "tensor_filter", "model", test_model2, NULL);
+
+  gst_harness_get (h, "tensor_filter", "model", &prop_string, NULL);
+  EXPECT_TRUE (g_str_equal (prop_string, test_model2));
+  g_free (prop_string);
+
+  /* push buffer again */
+  in_buf = gst_harness_create_buffer (h, in_size);
+  EXPECT_EQ (gst_harness_push (h, in_buf), GST_FLOW_OK);
+
+  /* get output buffer */
+  out_buf = gst_harness_pull (h);
+  EXPECT_EQ (gst_buffer_n_memory (out_buf), 1U);
+  EXPECT_EQ (gst_buffer_get_size (out_buf), out_size);
+  gst_buffer_unref (out_buf);
+
+  gst_harness_teardown (h);
+  g_free (test_model);
+  g_free (test_model2);
+}
+
+/**
+ * @brief Test to reload tf-lite; model does not exist (negative)
+ */
+TEST (test_tensor_filter, reload_tflite_model_not_found_n)
+{
+  const gchar fw_name[] = "tensorflow-lite";
+  const GstTensorFilterFramework *fw = nnstreamer_filter_find (fw_name);
+  GstTensorFilterProperties *prop = NULL;
+  gpointer private_data = NULL;
+
+  const gchar *root_path = g_getenv ("NNSTREAMER_BUILD_ROOT_PATH");
+  gchar *test_model;
+
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  test_model = g_build_filename (root_path, "tests", "test_models", "models",
+      "mobilenet_v1_1.0_224_quant.tflite", NULL);
+
+  ASSERT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  const gchar *model_files[] = {
+    test_model, NULL,
+  };
+
+  /* prepare properties */
+  prop = g_new0 (GstTensorFilterProperties, 1);
+  ASSERT_TRUE (prop != NULL);
+
+  prop->fwname = fw_name;
+  prop->model_files = model_files;
+  prop->num_models = 1;
+
+  ASSERT_TRUE (fw && fw->open && fw->close && fw->reloadModel);
+
+  /* open tf-lite model */
+  EXPECT_TRUE (fw->open (prop, &private_data) == 0);
+
+  g_free (test_model);
+  test_model = g_build_filename (root_path, "tests", "test_models", "models",
+      "mobilenet_v2_1.0_224_quant.tflite", NULL);
+  ((gchar **)model_files)[0] = test_model; /* remove const for the test */
+
+  /* reload tf-lite model */
+  EXPECT_TRUE (fw->reloadModel (prop, &private_data) == 0);
+
+  g_free (test_model);
+  test_model = g_build_filename (root_path, "tests", "test_models", "models",
+      "model_does_not_exist.tflite", NULL);
+  ((gchar **)model_files)[0] = test_model; /* remove const for the test */
+
+  /* reload tf-lite model which does not exist */
+  EXPECT_FALSE (fw->reloadModel (prop, &private_data) == 0);
+
+  /* close tf-lite model */
+  fw->close (prop, &private_data);
+
+  g_free (prop);
+  g_free (test_model);
+}
+
+/**
+ * @brief Test to reload tf-lite; model has wrong dimension (negative)
+ */
+TEST (test_tensor_filter, reload_tflite_model_wrong_dims_n)
```

Comment:
This comment is not directly related to this PR.
If the new model to be reloaded has different dimension, then the current implementation and test case makes the reloading fail.
However, if the new model to be reloaded allows changing the dimension, and if the dimension for the new model can be changed to the required dimension, then reload can be success.

Reply:
Actually, supporting different dimensions is not a primary target because we're targeting on-device training like changed weights in runtime. If someone wants to reload a new model with different dimensions, we can say like just relaunching the pipeline and re-performing caps navigation.

Reply:
Yes, the scenario I presented is different from on-device training.
In this case, the models default input dimensions are different, but can support the currently set input dimensions. Anyway, this can be considered later when needed.

Reply:
Sure. Let's do it step-by-step.